engine/src/vault/crypto_box.rs

```diff
@@ -51,37 +57,32 @@ impl<T: BoxProvider> Key<T> {
     /// generate a random key using secure random bytes
     pub fn random() -> Self {
         Self {
-            key: GuardedVec::new(T::box_key_len(), |v| {
-                v.copy_from_slice(
+            key: {
+                Buffer::alloc(
                     T::random_vec(T::box_key_len())
                         .expect("failed to generate random key")
                         .as_slice(),
+                    T::box_key_len(),
                 )
-            }),
-
+            },
             _box_provider: PhantomData,
         }
     }
 
     /// attempts to load a key from inputted data
     ///
     /// Return `None` if the key length doesn't match [`BoxProvider::box_key_len`].
+    #[allow(dead_code)]
     pub fn load(key: Vec<u8>) -> Option<Self> {
         if key.len() == T::box_key_len() {
             Some(Self {
-                key: GuardedVec::new(T::box_key_len(), |v| v.copy_from_slice(key.as_slice())),
+                key: Buffer::alloc(key.as_slice(), T::box_key_len()),
                 _box_provider: PhantomData,
             })
         } else {
             None
         }
     }
-
-    /// get the key's bytes from the [`GuardedVec`]
-    pub fn bytes(&self) -> Vec<u8> {
-        // hacks the guarded type.  Probably not the best solution.
-        (*self.key.borrow()).to_vec()
-    }
```

Comment:
Why is this method removed? I think we need such a convenience method again, to avoid that the user has to the awkward call of:
```
let key = *(&key.key).borrow()
```
to access the actual key.